### PR TITLE
Prevent crash from Billion Laughs attack.

### PR DESCRIPTION
**Pull request by JimBobSquarePants**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/Fonts/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

See

https://rawgit.com/unicode-org/text-rendering-tests/master/reports/fontkit.html#GSUB-3
https://en.wikipedia.org/wiki/Billion_laughs_attack
https://github.com/foliojs/fontkit/issues/221

### CODE REVIEW

File: tests/SixLabors.Fonts.Tests/Tables/AdvancedTypographic/Gsub/GSubTableTests.cs [hunk 1]
```diff
@@ -368,5 +368,15 @@ public void OldStyleFiguresFeature_Works()
...
         }
+
+        [Fact]
+        public void BillionLaughsAttackDoesNotThrowException()
+        {
+            // Arrange
+            Font font = new FontCollection().Add(TestFonts.GSubLookupType2BillionLaughs).CreateFont(12);
+
+            // Act
+            TextMeasurer.Measure("lol", new TextOptions(font));
```
Comment (brianpopow) [inline tests/SixLabors.Fonts.Tests/Tables/AdvancedTypographic/Gsub/GSubTableTests.cs:379]: While text measure will not throw hear (or hang indefinitely), it still gives a ridiculous large number for width, which can cause an exception from the memory allocator if the user try's to render this. Should we try to address this also or should the user be responsible for checking sane values of width and height before rendering?

Comment (JimBobSquarePants) [reply tests/SixLabors.Fonts.Tests/Tables/AdvancedTypographic/Gsub/GSubTableTests.cs:379]: Looks like I’m using the wrong constant. That one is for subsetting. Will update later.

Comment (JimBobSquarePants) [reply tests/SixLabors.Fonts.Tests/Tables/AdvancedTypographic/Gsub/GSubTableTests.cs:379]: Done. We check both the length and the number of operations for GSUB, GPOS, & KERN.

Comment (brianpopow) [reply tests/SixLabors.Fonts.Tests/Tables/AdvancedTypographic/Gsub/GSubTableTests.cs:379]: That's much better, also a good idea to use also in GPOS and KERN.

**Review by brianpopow — APPROVED** [review]

**Pull request merged**